gateleen-core/src/main/java/org/swisspush/gateleen/core/http/BufferBridge.java

```diff
@@ -67,11 +70,14 @@ protected void doWrite(Buffer chunk) {
 
     protected Future<Void> doEnd() {
         ended = true;
-        if(endHandler != null && queue.isEmpty()) {
+        if (bodyHandler != null) {
```

Comment:
Follows the same approach as in https://github.com/eclipse-vertx/vert.x/blob/4.2.1/src/main/java/io/vertx/core/http/impl/HttpEventHandler.java#L79-L82